### PR TITLE
Remove Provider Deprecations in Trino 

### PR DESCRIPTION
**Pull request by Prab-27**
Related: #44559 

Remove deprecated code from trino provider from operators and test. Update Changelog

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Can you fix the tests please ?

**Comment by Prab-27** [comment]
Would you please tell me how to fix this test ? 

There is only one deprecated class  `TrinoOperator` in [providers/src/airflow/providers/trino/operators/trino.py](url) 

When I fixed `mypy` and removed that files , they showed me deprecated module error 

To fix I have added that files again but but it failed here.

**Comment by potiuk** [comment]
This tests uses TrinoOperator:

```
    @mock.patch.dict("os.environ", AIRFLOW_CONN_TRINO_DEFAULT="trino://airflow@trino:8080/")
    def test_openlineage_methods(self):
        op = TrinoOperator(task_id="trino_test", sql="SELECT name FROM tpch.sf1.customer LIMIT 3")
        op.execute({})
        lineage = op.get_openlineage_facets_on_start()
        assert lineage.inputs[0].namespace == "trino://trino:8080"
        assert lineage.inputs[0].name == "tpch.sf1.customer"
        assert "schema" in lineage.inputs[0].facets
        assert lineage.job_facets["sql"].query == "SELECT name FROM tpch.sf1.customer LIMIT 3"

```

Replace with SQLExecuteQueryOperator -as suggested by deprecation message.

**Comment by Prab-27** [comment]
Done!

**Review by potiuk — APPROVED** [review]

File: providers/tests/trino/operators/test_trino.py [hunk 1]
```diff
@@ -1,113 +0,0 @@
...
-                        SchemaDatasetFacetFields(name="name", type="varchar(25)"),
-                        SchemaDatasetFacetFields(name="address", type="varchar(40)"),
-                        SchemaDatasetFacetFields(name="nationkey", type="bigint"),
-                        SchemaDatasetFacetFields(name="phone", type="varchar(15)"),
-                        SchemaDatasetFacetFields(name="acctbal", type="double"),
-                    ]
-                )
-            },
-        )
-    ]
```
Comment (eladkal) [inline providers/tests/trino/operators/test_trino.py:109]: I am not sure if this was suppose to be removed?
this is testing open lineage integration with `TrinoHook`
cc @kacpermuda @mobuchowski

Comment (kacpermuda) [reply providers/tests/trino/operators/test_trino.py:109]: Yes, i think it should be adjusted after the removal of deprecated code and not fully removed.

Comment (mobuchowski) [reply providers/tests/trino/operators/test_trino.py:109]: Yeah, this was probably removed by mistake.

@Prab-27 can you revert that test?

Comment (kacpermuda) [reply providers/tests/trino/operators/test_trino.py:109]: It was probably removed as the `operators/trino.py` file has been removed, so it makes sense that the corresponding test file is removed, but IMHO we should still keep this test in the trino provider (maybe in a differently named file).

File: providers/tests/integration/trino/hooks/test_trino.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def test_should_record_records_with_kerberos_auth(self):
 
     @mock.patch.dict("os.environ", AIRFLOW_CONN_TRINO_DEFAULT="trino://airflow@trino:8080/")
     def test_openlineage_methods(self):
-        op = TrinoOperator(task_id="trino_test", sql="SELECT name FROM tpch.sf1.customer LIMIT 3")
+        op = SQLExecuteQueryOperator(task_id="trino_test", sql="SELECT name FROM tpch.sf1.customer LIMIT 3")
```
Comment (kacpermuda) [inline providers/tests/integration/trino/hooks/test_trino.py:53]: we should add `conn_id="trino_default"` as this test is failing now.

Comment (Prab-27) [reply providers/tests/integration/trino/hooks/test_trino.py:53]: You are right it was removed by mistake.

Comment (Prab-27) [reply providers/tests/integration/trino/hooks/test_trino.py:53]: Ya I'll correct that id

**Comment by Prab-27** [comment]
I am Sorry 
Could you please tell me where I can add this test code ?

[providers/tests/trino/hooks/test_trino.py](url) ? here ?

**Comment by eladkal** [comment]
> [providers/tests/trino/hooks/test_trino.py](https://github.com/apache/airflow/pull/url) ? here ?

That would be ok

**Comment by Prab-27** [comment]
Would you please tell me which test I should run to check if it is okay or not ? 

Does this work ? 
` breeze testing providers-tests --test-type "Providers[trino]"`

it passed at that time

**Comment by kacpermuda** [comment]
Ah, sorry you meant the other test. Yes, the breeze command looks good.

**Comment by Prab-27** [comment]
No, you are right. I am asking about the provider test, not Breeze, so I can know about that.
The Breeze tests failed due to my mistakes. 
Sorry, I am a newbie, so I don’t know about all this. I am trying to learn

**Comment by Prab-27** [comment]
I am confused. Would You please tell me if add this method here 

[providers/tests/integration/trino/hooks/test_trino.py](url)  because this file contains similar methods
or [[providers/tests/trino/hooks/test_trino.py]](url)

**Comment by eladkal** [comment]
I'm not sure I follow the issue.
The test this PR removed was in `providers/tests/trino/operators/test_trino.py`
So we just need to bring it back, the best candidate is `providers/tests/trino/hooks/test_trino.py`

Why are we discussing system tests path?

**Comment by Prab-27** [comment]
Sorry , You are right I noticed similar method `test_openlineage_methods`  at here so I asked [providers/tests/integration/trino/hooks/test_trino.py](url)

**Pull request merged**